File: packages/intune/ios/Plugin/Classes/Results/AcquireTokenResult.swift
import Foundation
import Capacitor

@objc public class AcquireTokenResult: NSObject, Result {
    let accessToken: String
    let accountId: String
    let idToken: String?
    let tenantId: String?
    let username: String?

    init(accessToken: String, accountId: String, idToken: String?, tenantId: String?, username: String?) {
        self.accessToken = accessToken
        self.accountId = accountId
        self.idToken = idToken
        self.tenantId = tenantId
        self.username = username
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["accessToken"] = accessToken
        result["accountId"] = accountId
        result["idToken"] = idToken == nil ? NSNull() : idToken
        result["tenantId"] = tenantId == nil ? NSNull() : tenantId
        result["username"] = username == nil ? NSNull() : username
        return result as AnyObject
    }
}

File: packages/intune/android/src/main/java/io/capawesome/capacitorjs/plugins/intune/classes/results/AcquireTokenResult.java
package io.capawesome.capacitorjs.plugins.intune.classes.results;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.intune.interfaces.Result;
import org.json.JSONObject;

public class AcquireTokenResult implements Result {

    @NonNull
    private final String accessToken;

    @NonNull
    private final String accountId;

    @Nullable
    private final String idToken;

    @Nullable
    private final String tenantId;

    @Nullable
    private final String username;

    public AcquireTokenResult(
        @NonNull String accessToken,
        @NonNull String accountId,
        @Nullable String idToken,
        @Nullable String tenantId,
        @Nullable String username
    ) {
        this.accessToken = accessToken;
        this.accountId = accountId;
        this.idToken = idToken;
        this.tenantId = tenantId;
        this.username = username;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("accessToken", accessToken);
        result.put("accountId", accountId);
        result.put("idToken", idToken == null ? JSONObject.NULL : idToken);
        result.put("tenantId", tenantId == null ? JSONObject.NULL : tenantId);
        result.put("username", username == null ? JSONObject.NULL : username);
        return result;
    }
}
